pkg/apis/serving/v1alpha1/kfservice_types.go

```diff
@@ -42,6 +41,29 @@ type ModelSpec struct {
 	XGBoost    *XGBoostSpec    `json:"xgboost,omitempty"`
 	SKLearn    *SKLearnSpec    `json:"sklearn,omitempty"`
 	PyTorch    *PyTorchSpec    `json:"pytorch,omitempty"`
+	Explain    *ExplainerSpec  `json:"explain,omitempty"`
+}
+
+// ExplainerSpec defines the arguments for a model explanation server
+type ExplainerSpec struct {
+	// The following fields follow a "1-of" semantic. Users must specify exactly one spec.
+	Alibi  *AlibiExplainerSpec `json:"alibi,omitempty"`
+	Custom *CustomSpec         `json:"custom,omitempty"`
+}
+
+// AlibiExplainerSpec defines the arguments for configuring an Alibi explanation serving
+type AlibiExplainerSpec struct {
+	// The following fields follow a "1-of" semantic. Users must specify exactly one of Type or ExplainerURI.
+	// The kind of Alibi explanation server to create, e.g. AnchorTabular
+	Type string `json:"type"`
+	// The location of a fit explanation model
+	SavedExplainerURI string `json:"savedExplainerUri,omitempty"`
```

Comment:
is SavedExplainerURI used to point an existing explainer service url?

Reply:
No, to a saved explainer model. So simialr to predictors that can load their modelWeights from a bucket so can an explainer.

Reply:
I see. Thanks.